packages/grid/_modules_/grid/models/api/columnMenuApi.ts

```diff
@@ -5,8 +5,10 @@ export interface ColumnMenuApi {
   /**
    * Display the column menu under the field column.
    * @param field
+   * @param id
+   * @param labelledby
    */
-  showColumnMenu: (field: string) => void;
+  showColumnMenu: (field: string, id: string, labelledby: string) => void;
```

Comment:
why do we need id and labelledBy to show the column menu? And how can consumers get them?

Reply:
we need them so that the `labelledby` and the `controlledby` are hooked up correctly. I had to do it like this because they that the menu and the button are set up that is the only option. In terms of how users can get them -> there are in the state while the menu is opened.

Reply:
ShowColumnMenu should only show the menu when one click on the 3 dots icon next to the column title.
Is that still the case?

> there are in the state while the menu is opened.

Well `showColumnMenu` opens the menu so how can we know the `labelledBy` before we call the `showColumnMenu`. Chicken and egg situation? :)

Reply:
I mean the existing functionality still works the same.

Reply:
@dtassone can I merge this, it would be nice to include the fix in the upcoming release 🙃

Reply:
I can approve if you want, but for me the api is not great. I have to pass a mandatory Id that I can't know, and a labelled by that I can't know either. I don't see how this change is going to improve the DX and api.

Reply:
> why do we need id and labelledBy to show the column menu?

@dtassone The answer is in https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

> And how can consumers get them?

I think that the question should be the opposite, why is this method public in the first place?

Reply:
how do you render the grid with the menu open. Imagine if the user want to put a custom component...

Reply:
How is it harmful?

Reply:
I don't think that would be an issue, they can still open it right, it's just that the `id` and the `labelledBy` won't be set

Reply:
> How is it harmful?

It could prevent us to improve the internal: e.g. this very discussion. I don't understand the use cases for a developer to open a column menu from outside the normal end-user flows.

Reply:
> I don't think that would be an issue, they can still open it right, it's just that the `id` and the `labelledBy` won't be set

why not make them optional?
Also why calling it id? Something like menuElementId or whatever a bit more specific so it's more understandable